tests/test_send_recv_two_workers.py

```diff
@@ -0,0 +1,175 @@
+import os
+import asyncio
+import pytest
+import pickle
+import time
+
+import ucp
+import struct
+import multiprocessing
+from distributed.utils import nbytes
+
+
+async def get_ep(name, port):
+    addr = ucp.get_address().encode()
+    ep = await ucp.get_endpoint(addr, port)
+    print(name, ep)
+    return ep
+
+
+def client(env, port):
+    # wait for server to come up
+    # receive cudf object
+    # deserialize
+    # assert deserialized msg is cdf
+    # send receipt
+
+    os.environ.update(env)
+    ucp.init()
+
+    async def read():
+        await asyncio.sleep(3)
+        ep = await get_ep("client", port)
+        print("Waiting to receiving frames...")
+
+        resp = await ep.recv_future()
+        obj = ucp.get_obj_from_msg(resp)
+        nframes, = struct.unpack("Q", obj[:8])  # first eight bytes for number of frames
+        gpu_frame_msg = obj[
+            8 : 8 + nframes
+        ]  # next nframes bytes for if they're GPU frames
+        is_gpus = struct.unpack("{}?".format(nframes), gpu_frame_msg)
+
+        sized_frame_msg = obj[8 + nframes :]  # then the rest for frame sizes
+        sizes = struct.unpack("{}Q".format(nframes), sized_frame_msg)
+
+        frames = []
+
+        for i, (is_gpu, size) in enumerate(zip(is_gpus, sizes)):
+            if size > 0:
+                resp = await ep.recv_obj(size, cuda=is_gpu)
+            else:
+                resp = await ep.recv_future()
+            frame = ucp.get_obj_from_msg(resp)
+            frames.append(frame)
+
+        header = pickle.loads(frames[0])
+        frames = frames[1:]  #
+        typ = pickle.loads(header["type"])
+        cuda_obj = typ.deserialize(header, frames)
+
+        print(cuda_obj, len(cuda_obj))
+        asyncio.sleep(5)
+        # how should we confirm data is on two GPUs ?
```

Comment:
@kkraus14 / @pentschev  at this point in the test we have sent data from GPU1 to GPU5.  Do you know of an easy way to confirm data on both GPUs ?

Reply:
You might want to refer to this pynvml code that we use within distributed to get the memory use and utilization of each GPU: https://github.com/dask/distributed/blob/master/distributed/diagnostics/nvml.py

Reply:
You can also use `gdf['a'].data.mem.gpu_data.context.PCI_BUS_ID` which will return the unique PCI bus identifier of the GPU (will not be 0 - 7 on a DGX for example)